acceptance-tests/test-plugins/src/test/java/org/hyperledger/besu/services/BesuPluginContextImplTest.java

```diff
@@ -158,9 +172,68 @@ public void lifecycleExceptions() throws Throwable {
     assertThatExceptionOfType(IllegalStateException.class).isThrownBy(contextImpl::stopPlugins);
   }
 
-  private Optional<TestPicoCLIPlugin> findTestPlugin(final List<BesuPlugin> plugins) {
+  @Test
+  public void shouldRegisterAllPluginsWhenNoPluginsOption() {
+    final PluginConfiguration config = createConfigurationForAllPlugins();
+
+    assertThat(contextImpl.getRegisteredPlugins()).isEmpty();
+    contextImpl.registerPlugins(config);
+    final Optional<TestPicoCLIPlugin> testPluginOptional =
+        findTestPlugin(contextImpl.getRegisteredPlugins(), TestPicoCLIPlugin.class);
+    Assertions.assertThat(testPluginOptional).isPresent();
+    final TestPicoCLIPlugin testPicoCLIPlugin = testPluginOptional.get();
+    assertThat(testPicoCLIPlugin.getState()).isEqualTo("registered");
+  }
+
+  @Test
+  public void shouldRegisterOnlySpecifiedPluginWhenPluginsOptionIsSet() {
+    final PluginConfiguration config = createConfigurationForSpecificPlugin("TestPicoCLIPlugin");
+
+    assertThat(contextImpl.getRegisteredPlugins()).isEmpty();
+    contextImpl.registerPlugins(config);
+
+    final Optional<TestPicoCLIPlugin> nonRequestedPlugin =
+        findTestPlugin(contextImpl.getRegisteredPlugins(), TestBesuEventsPlugin.class);
+
+    Assertions.assertThat(nonRequestedPlugin).isEmpty();
```

Comment:
I think this test is missing the check that the requested plugin is registered, otherwise is the same as the test below

Reply:
done